### PR TITLE
Move proxy as optional permission/availabe in the options page

### PR DESCRIPTION
**Pull request by maxxcrawford**
This PR does the following: 
- Moves proxy to optional permission in the manifest
- Adds a checkbox in the options page to toggle this permission/feature
- Show/hide the proxy settings in the container mgmt panel based on the permission status

### CODE REVIEW

**Comment by maxxcrawford** [comment]
cc/ @kendallcorner

File: src/js/background/messageHandler.js [hunk 1]
```diff
@@ -23,6 +23,9 @@ const messageHandler = {
       case "resetBookmarksContext":
         response = assignManager.resetBookmarksMenuItem();
         break;
+      case "resetProxySupport":
+        response = assignManager.init();
```
Comment (groovecoder) [inline src/js/background/messageHandler.js:27]: Is this the right UX?

1. Enable proxy support
2. Assign Container1 to use proxy1
3. Open sensitiveSite1 in Container1 (thru proxy1)
4. Disable proxy support
5. Reload tab from step 3
   * sensitiveSite1 is reloaded without a proxy without any other warning

**Review by groovecoder — CHANGES_REQUESTED** [review]

File: src/js/popup.js [hunk 2]
```diff
@@ -1428,6 +1428,16 @@ Logic.registerPanel(P_CONTAINER_EDIT, {
...
     });
 
+    // Query if user has proxy permissions. 
+    // If so, we want to show the custom proxy settings options for the container: 
+    const hasProxyPermission = await browser.runtime.sendMessage({
+      method: "proxyPermissionCheck",
+    });
+
+    if (hasProxyPermission) {
+      document.querySelector(".proxy-container-settings").classList.remove("is-hidden");
```
[the remark this answers is not in the document]
Comment (maxxcrawford) [reply src/js/popup.js:1438]: Yes – This is just a framework for @lesleyjanenorton to key off of.

**Comment by maxxcrawford** [comment]
Closing this for now.

**Pull request closed without merging**